kw6/stream.py

```diff
@@ -45,7 +45,37 @@ def from_path(path, header_path=None):
             cached_positions=header_positions(header_path) if header_path is not None else {},
         )
 
-    def seek_(self, frame_index: types.FRAME_INDEX):
+    def __getitem__(self, indices_or_slice):
+        if type(indices_or_slice) == int:
+            if indices_or_slice < 0:
+                raise IndexError('Negative indexing not supported.')
+            self._seek(indices_or_slice)
+            if not self.empty():
+                positions = Position.from_stream_(self.stream)
+            else:
+                raise IndexError(indices_or_slice)
+
+        elif type(indices_or_slice) == slice:
+            if indices_or_slice.start is None:
```

Comment:
Felt hard to read the control flow with
`if ... if ... else ...`

Maybe you can skip the first `if` and do something like
```
for index in range(indices_or_slice.start || 0, indices_or_slice.stop)
```

Reply:
That doesn't handle the case when `indices_or_slice.stop` is None. I agree that the current control flow is hard to read though and would like to come up with a cleaner solution, will think on it for a bit

Reply:
The stop case is fine as is. Just remove the first if. You would just keep the `if ... else ...`

Reply:
`self.seek_(indices_or_slice.start)` on line 79 would fail but I guess I can change that one to
```
self.seek_(
    indices_or_slice.start
    if indices_or_slice.start is not None
    else 0
)
```
unless you have a cleaner solution?

Reply:
Ill just do it at the top instead.
```
start = (
    indices_or_slice.start
    if indices_or_slice.start is not None
    else 0
)
```

Reply:
Done